src/async.rs

```diff
@@ -455,4 +424,26 @@ impl AsyncClient {
     pub fn client(&self) -> &Client {
         &self.client
     }
+
+    /// Sends a GET request to the given `url`, retrying failed attempts
+    /// for retryable error codes until max retries hit.
+    async fn get_with_retry(&self, url: &str) -> Result<Response, Error> {
+        let mut delay = BASE_BACKOFF_MILLIS;
+        let mut attempts = 0;
+
+        loop {
+            match self.client.get(url).send().await? {
+                resp if attempts <= self.max_retries && is_status_retryable(resp.status()) => {
```

Comment:
I think this was probably better using `<` rather than `<=`

Reply:
I don't see much problem using `<=`, although indeed using `<=` would add a few more seconds as it "adds another attempt".

Reply:
However, for the sake of correctness with the used constant, I'm fine we can update it `<`.